src/lib/select/select.spec.ts

```diff
@@ -2690,8 +2693,74 @@ describe('MdSelect', () => {
 
   });
 
-});
+  describe('compareWith behavior', () => {
+    let fixture: ComponentFixture<NgModelCompareWithSelect>;
+    let instance: NgModelCompareWithSelect;
+
+    beforeEach(async(() => {
+      fixture = TestBed.createComponent(NgModelCompareWithSelect);
+      instance = fixture.componentInstance;
+      fixture.detectChanges();
+    }));
+
+    it('should have a selection', () => {
+      const selectedOption = instance.select.selected as MdOption;
+      expect(selectedOption.value.value).toEqual('pizza-1');
+    });
+
+    it('should update when making a new selection', async(() => {
+      instance.options.last._selectViaInteraction();
+      fixture.detectChanges();
+      fixture.whenStable().then(() => {
+        const selectedOption = instance.select.selected as MdOption;
+        expect(instance.selectedFood.value).toEqual('tacos-2');
+        expect(selectedOption.value.value).toEqual('tacos-2');
+      });
+    }));
+
+    describe('when comparing by reference', () => {
+      beforeEach(async(() => {
+        spyOn(instance, 'compareByReference').and.callThrough();
+        instance.useCompareByReference();
+        fixture.detectChanges();
+      }));
+
+      it('should use the comparator', () => {
+        expect(instance.compareByReference).toHaveBeenCalled();
+      });
+
+      it('should initialize with no selection despite having a value', () => {
+        expect(instance.selectedFood.value).toBe('pizza-1');
+        expect(instance.select.selected).toBeUndefined();
+      });
+
+      it('should not update the selection when changing the value', async(() => {
```

Comment:
Nit: I think this test name could be misleading. Selects comparing by reference should still update the selection, generally speaking. It's only when you manually copy over every ngModelChange that you see this behavior.  Maybe: "should not update the selection if value is copied on change" ?